packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/platform/SecureRandom.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Realm Inc.
+ * Copyright 2017-2019 Carlos Ballesteros Velasco and contributors
+    * https://github.com/korlibs/korge/graphs/contributors
+    * https://github.com/korlibs-archive/
+ * All rights reserved.
+
+ * MIT License
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+
+ * See: https://github.com/korlibs/korge/tree/main/krypto
+*/
+
+package io.realm.kotlin.internal.platform
+
+import kotlin.random.Random
+
+internal expect fun fillRandomBytes(array: ByteArray)
+internal expect fun seedExtraRandomBytes(array: ByteArray)
+
+internal fun arraycopy(src: ByteArray, srcPos: Int, dst: ByteArray, dstPos: Int, count: Int) = src.copyInto(dst, dstPos, srcPos, srcPos + count)
+
+@Suppress("MagicNumber")
+internal object SecureRandom : Random() {
+    fun addSeed(array: ByteArray) {
+        seedExtraRandomBytes(array)
+    }
+
+    private fun getInt(): Int {
+        val temp = ByteArray(4)
+        fillRandomBytes(temp)
+        val a = temp[0].toInt() and 0xFF
+        val b = temp[1].toInt() and 0xFF
+        val c = temp[2].toInt() and 0xFF
+        val d = temp[3].toInt() and 0xFF
+        return (a shl 24) or (b shl 16) or (c shl 8) or (d shl 0)
+    }
+
+    override fun nextBytes(array: ByteArray, fromIndex: Int, toIndex: Int): ByteArray {
```

Comment:
Any reason we are overriding this? `Random` has an implementation that fallback to `nextBits`...but maybe this is just more efficient?

Reply:
I largely just went with Korge's implementation so did not make this decision myself, but yeah I imagine it might be slightly more efficient as I suppose they might be making this nextBytes-centric(?) since this is what gets used more for their (and our?) cases?

Reply:
Hm, not sure why but removing `nextBytes` override does lead to errors, I am not sure why the fallback to `nextBits` is not working because it does run those lines but ends up with just empty lists

Reply:
You cannot have a `fun nextBytes` method in SecureRandom if `Random` contains the same method. Was that what you where seeing?

Reply:
In any case, I think having this is fine, as long as we test it.

Reply:
Nope, I was trying to remove the function all together so it would inherit the `Random.nextBytes` but I think something would go wrong with that. It'd run the `nextBytes` code from `Random` but just give out 0s. I will look into this for a little more just to make sure this is not a problem with `nextBits` it uses.

Reply:
Yeah, that definitely sounds like something that needs to be looked into 🤔

Reply:
Yeah, I think something was wrong: https://github.com/realm/realm-kotlin/pull/1146#discussion_r1036924404